scrapy/utils/item.py

```diff
@@ -0,0 +1,104 @@
+from collections.abc import MutableMapping
+
+from scrapy.item import BaseItem
+
+
+def _is_dataclass_instance(obj):
+    """
+    Returns True if *obj* is a dataclass object, False otherwise.
```

Comment:
a nitpick: as per [pep 257](https://www.python.org/dev/peps/pep-0257/), it should be "Return ...", not "Returns ...":

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".